tsl/src/continuous_aggs/create.c

```diff
@@ -2863,6 +2865,16 @@ tsl_process_continuous_agg_viewstmt(Node *node, const char *query_string, void *
 
 		continuous_agg_refresh_internal(cagg, &refresh_window, CAGG_REFRESH_CREATION, true, true);
 	}
+	else
+	{
+		/* Set NULL watermark */
+		Hypertable *ht = ts_hypertable_get_by_id(cagg->data.mat_hypertable_id);
+		Assert(NULL != ht);
```

Comment:
Since this is reading from the catalog, you might want to use `Ensure` here to not crash in production builds if the catalog data is corrupt.

Reply:
Fixed!